cpp/src/arrow/flight/test_util.cc

```diff
@@ -203,6 +203,37 @@ class FlightTestServer : public FlightServerBase {
     return Status::OK();
   }
 
+  Status DoExchange(const ServerCallContext& context,
+                    std::unique_ptr<FlightMessageReader> reader,
+                    std::unique_ptr<FlightMessageWriter> writer) override {
+    std::vector<std::shared_ptr<RecordBatch>> batches;
+    FlightStreamChunk chunk{};
+    int chunks = 0;
+    while (true) {
+      RETURN_NOT_OK(reader->Next(&chunk));
+
+      if (chunk.data) {
+        batches.push_back(chunk.data);
+      }
+
+      if (!chunk.data && !chunk.app_metadata) {
+        break;
+      }
+      chunks++;
+    }
+
+    std::shared_ptr<Buffer> buf;
+    RETURN_NOT_OK(Buffer::FromString(std::to_string(chunks), &buf));
+    RETURN_NOT_OK(writer->WriteMetadata(buf));
```

Comment:
Add a comment what this is about?

Reply:
I've clarified what each of the test cases look for (and added more test cases).